src/store/reducers/index.ts

```diff
@@ -77,6 +74,11 @@ export const rootReducer = {
     fullscreen,
 };
 
-export default combineReducers({
+const combinedReducer = combineReducers({
     ...rootReducer,
 });
+
+export type IRootReducer = typeof combinedReducer;
+export type IRootState = ReturnType<IRootReducer>;
```

Comment:
I did. 

The problem is in state-url-mapping enhancer, that expands store with its own data and reduces store type to any. To prevent it this function needs types as well. However, it is quite complex thing that uses redux-location-state lib, that does not have its own types.

So, I decided not to spend more time and produce potential merge conflicts, but stick to this solution, as it is sufficient for solving current problems (except for typed dispatch).

I suppose, the complete store typing could be done in next PRs.